rlpx/rlpx.go

```diff
@@ -175,9 +175,17 @@ func (s *session) decode(buf []byte) (byte, rlp.Item, error) {
 		return 0, rlp.Item{}, errors.New("invalid frame mac")
 	}
 	s.ig.stream.XORKeyStream(frame, frame)
-	code := frame[0:1]
+
+	code, err := rlp.Decode(frame[0:1])
```

Comment:
I wonder if we should decode the entire frame here since technically the msg-id could be more than a single byte.

Reply:
ah. good call. strange